src/coreclr/gc/gc.cpp

```diff
@@ -23480,8 +23484,19 @@ void gc_heap::background_mark_simple1 (uint8_t* oo THREAD_NUMBER_DCL)
                 else
                 {
                     dprintf (3,("mark stack overflow for object %Ix ", (size_t)oo));
+#ifdef USE_REGIONS
+                    heap_segment* overflow_region = region_of (oo);
+                    overflow_region->flags |= heap_segment_flags_overflow;
+                    gc_heap* overflow_heap = heap_of (oo);
+                    overflow_heap->background_overflow_p = TRUE;
+                    if (heap_segment_gen_num (overflow_region) < max_generation)
+                    {
+                        overflow_heap->background_eph_overflow_p = TRUE;
```

Comment:
>background_eph_overflow_p = TRUE; [](http://example.com/codeflow?start=39&length=33)

you don't need to set eph overflow here... by the time we process OF, those may not even be in the ephemeral generations anymore. you could detect this in `background_process_mark_overflow`